### ISSUE TITLE
4.9 regression: unexpected error using in operator on generic object intersected with `{}`

### ISSUE THREAD

**Issue by OliverJAsh** [microsoft/TypeScript#51549]
# Bug Report

### 🔎 Search Terms

- `in` operator
- generic
- object

### 🕗 Version & Regression Information

- This changed between versions 4.8 and 4.9

### ⏯ Playground Link

[Playground link with relevant code](https://www.typescriptlang.org/play?ts=5.0.0-dev.20221116#code/MYewdgzgLgBAZjAvDAPABRgUwB5U2AEwhhACMArTYKAPgAoBDALhgwDIYBvAXwEokaXAFAxRMAPTiYAFQCeAB0wwA5Oy7dlMALYNZMAE6Z5hiPlgMYxgJZarUKwDclDhgBsArpgA0MAO4ALK2B-GCtiMBBYRX1bKDwCGAZiKH8lfSsAc39YEGiGQhIEFKVlKzBNXMx9BigQfQA6KAg6ACYANgBmAA5eETEAIjgQEH7QsESAbiFuKaA)

### 💻 Code

```ts
const f = <P extends object>(a: P & {}) => {
    // Unexpected error:
    // Type 'P & {}' may represent a primitive value, which is not permitted as the right operand of the 'in' operator.ts(2638)
    "foo" in a;
};

```

### 🙁 Actual behavior

See code comment above.

### 🙂 Expected behavior

See code comment above.

**Comment by fatcerberus** [comment]
Likely a duplicate of #51501

**Comment by Andarist** [comment]
I can confirm that my PR fixes this issue. I've pushed out this repro case as an additional test case: https://github.com/microsoft/TypeScript/pull/51502/commits/42ac98e7c4e72280213ad10c74bd4f4f80d9c335
